### DISCUSSION TITLE
报错提示未检测到探针

### DISCUSSION THREAD

**Discussion by gweichao** [harmonycloud/kindling#643] [General]
1）环境, 
NAME   STATUS   ROLES  VERSION   INTERNAL-IP  OS,KERNEL-VERSION    CONTAINER-RUNTIME
vm20   Ready    master   v1.19.9   192.168.5.20 CentOS7,3.10.0-1160  docker://19.3.15
vm21   Ready    <none> v1.19.9   192.168.5.21 CentOS7,5.6.5-1         docker://19.3.15
vm22   Ready    <none> v1.19.9   192.168.5.22 CentOS7,5.6.5-1         docker://19.3.15
2) 已经安装 kindling-agent,v0.91; 下载运行skywalking7.0.0的oap、web-service, demo-app, agent。运行demo加入agent、oap、web-service。
skywalking是非容器版
3）已经下载编译 async-profiler(1.0.2)、kindling-java-agent(1.0.3)。尝试运行 ./profiler.sh start demo的oid、./profiler.sh start oap的pid，均提示如下类似的信息
		[Exist File] /proc/48071/root/tmp/kindling
		[Exist File] /proc/48071/root/tmp/kindling/1.0.3/libasyncProfiler.so
		start,version=1.0.3,file=/tmp/async-profiler.49660.48071
		2024-08:29 11:47:15 [x Attach] Aleady attached, no need to attach one more time.
4) 在浏览器访问skywalking正常，访问 kindling-camera的 http://192.168.5.21:9504，选择一个服务，总是提示 未检测到trace探针。
在官网、github、网络、问chatgpt都没有明确的帮助说明。

需要怎么搭建这部分的环境？
![2024-08-29 12 08 59](https://github.com/user-attachments/assets/d7357ac5-bf76-4762-8516-6e22a243e605)

**Comment by dxsup** [comment]
看到这种容器是Prometheus，是Go语言程序，固定会出现这个提示。但是不影响使用，点击“复杂视图”可以看到profiling。
能否在应用列表里找到oap和demo？这两个程序也会出现这个问题吗

**Comment by gweichao** [comment]
skywalking agent 有报错，我先排查一下：
WARN AbstractClassEnhancePluginDefine : enhance class cn.iocoder.springboot.lab39.skywalkingdemo.controller.DemoController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.RestControllerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class cn.iocoder.springboot.lab39.skywalkingdemo.controller.DemoController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.RestControllerForLowVersionInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.ControllerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.ControllerForLowVersionInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.ControllerInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.HandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.HandlerMethodInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.HandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.HandlerMethodInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.support.InvocableHandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.InvocableHandlerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.support.InvocableHandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.InvocableHandlerInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.

**Comment by gweichao** [comment]
界面上没有看到demoapp的服务，这个问题一直卡在那里。

skywalking agent 有报错，我先排查一下：
WARN AbstractClassEnhancePluginDefine : enhance class cn.iocoder.springboot.lab39.skywalkingdemo.controller.DemoController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.RestControllerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class cn.iocoder.springboot.lab39.skywalkingdemo.controller.DemoController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.RestControllerForLowVersionInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.ControllerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.ControllerForLowVersionInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.ControllerInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.HandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.HandlerMethodInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.HandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.HandlerMethodInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.support.InvocableHandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.InvocableHandlerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.support.InvocableHandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.InvocableHandlerInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.

**Comment by gweichao** [comment]
界面上没有看到demoapp的服务，这个问题一直卡在那里。

skywalking agent 有报错，我先排查一下：
WARN AbstractClassEnhancePluginDefine : enhance class cn.iocoder.springboot.lab39.skywalkingdemo.controller.DemoController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.RestControllerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class cn.iocoder.springboot.lab39.skywalkingdemo.controller.DemoController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.RestControllerForLowVersionInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.ControllerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.ControllerForLowVersionInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.ControllerInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.HandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.HandlerMethodInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.HandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.HandlerMethodInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.support.InvocableHandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v4.define.InvocableHandlerInstrumentation is not working. Because witness class org.springframework.cache.interceptor.DefaultKeyGenerator is not existed.
WARN AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.support.InvocableHandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.InvocableHandlerInstrumentation is not working. Because witness class org.springframework.web.servlet.view.xslt.AbstractXsltView is not existed.

**Comment by dxsup** [comment]
可以试试卸载skywalking agent，然后用页面上的“立即自动安装”

**Comment by gweichao** [comment]
我是直接选择、下载skywalking发布版本，跑 demoapp + agent、oam、web-ui服务，没有使用容器方式。这样不能跟kindling对接吗？
demoapp启动参数

/home/install/jdk/jdk1.8.0_381/bin/java -Dskywalking.agent.service_name=demoapp-lab-39 -Dskywalking.collector.backend_service=127.0.0.1:11800 -javaagent:/home/install/skywalking/agent-8.10.0/skywalking-agent.jar -jar lab-39-demo-2.2.2.RELEASE.jar

![demoapp,2024-08-29 20 26 30](https://github.com/user-attachments/assets/19ed6c15-2327-479a-88ce-a228d1ed38ca)

skywalking那边是可以看到请求业务的
![2024-08-29 20 23 13](https://github.com/user-attachments/assets/1c43874b-a6ba-4305-95fc-c03f647480b6)
![2024-08-29 20 23 49](https://github.com/user-attachments/assets/14e1d72f-6bd3-44d1-9185-df0a7483142f)
![2024-08-29 20 24 07](https://github.com/user-attachments/assets/87ffc45a-9ecb-457c-a7fe-2c67718b6dec)

**Comment by dxsup** [comment]
kindling这边是根本看不到这个服务，还是能找到这个服务但是仍然提示没有安装探针？

**Reply by gweichao** [reply]
根本看不到，估计是kindling需要容器方式安装的skywalking：
![k-2024-08-30 09 51 39](https://github.com/user-attachments/assets/6ded9179-da11-49d0-98d0-dd11c26f77e4)

**Comment by gweichao** [comment]
花了几天研究部署skywalking 容器版，后端8.9.1，agent 8.9.0
kindling还是看不到请求，而且提示未检测到tracing探针。

详见截图。需要给skyw挂一个 skywalking java agent吗？

![k-2024-09-05 18 01 06](https://github.com/user-attachments/assets/49ad153a-34dc-4240-aeb4-66a3e46f3e53)

**Comment by dxsup** [comment]
不需要，只要skywalking安装上去了，能连通OAP就行。
kindling里面应该能看到lab-demo才对，不行的话，重启kindling-agent试一下。

**Comment by gweichao** [comment]
下面这个正确了吧？应该点击第一个？

skywalking解决初始化失败、elasticsearch集群异常/分片超限等问题后，通过sidecar加载demo镜像，临时在demo的容器建立隧道连接oap，现在看到demo，点击也不提示检测不到tracing探针了。
![k-2024-09-11 10 54 46](https://github.com/user-attachments/assets/718e369d-2a93-4920-b9be-b891413e45ed)

**Comment by dxsup** [comment]
这个对了，然后出现这种空的情况时，可以尝试重启一下kindling-agent，然后再发起请求看下数据
